### PR TITLE
add kafka input support

### PR DESCRIPTION
**Pull request by McEchean**
add kafka input support

### CODE REVIEW

File: input/kafka/inputkafka.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	"syscall"
+)
+
+// ModuleName is the name used in config file
+const ModuleName = "kafka"
+
+// InputConfig holds the configuration json fields and internal objects
+type InputConfig struct {
+	config.InputConfig
+	Version      string   `json:"version"`       //Kafka cluster version, eg: 2.1.1
```
Comment (tengattack) [inline input/kafka/inputkafka.go]: We should add a space after `//`, like `// InputConfig holds ...`

File: input/kafka/inputkafka.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+			// recreated to get the new claims
+			if err := client.Consume(ctx, t.Topics, &cum); err != nil {
+				goglog.Logger.Errorf("Error from consumer: %v", err)
+			}
+			// check if context was cancelled, signaling that the consumer should stop
+			if ctx.Err() != nil {
+				return
+			}
+
+			cum.ready = make(chan bool)
```
Comment (tsaikd) [inline input/kafka/inputkafka.go]: Why renew the ready channel? (just curious)

File: input/kafka/inputkafka.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	// Do not move the code below to a goroutine.
+	// The `ConsumeClaim` itself is called within a goroutine, see:
+	// https://github.com/Shopify/sarama/blob/master/consumer_group.go#L27-L29
+	for message := range claim.Messages() {
+		//goglog.Logger.Printf("Message claimed: value = %s, timestamp = %v, topic = %s", string(message.Value), message.Timestamp, message.Topic)
+		var extra = map[string]interface{}{
+			"topic": message.Topic,
+			"timestamp": message.Timestamp,
+		}
+		ok, err := c.i.Codec.Decode(context.TODO(), string(message.Value), extra, []string{}, c.ch)
```
Comment (tsaikd) [inline input/kafka/inputkafka.go]: Why use `context.TODO()`?
1. why not main context?
2. why not `context.Background()`?

File: input/kafka/inputkafka.go (hunk 2 above)
Comment (McEchean) [reply input/kafka/inputkafka.go]: the example of sarama written like this,I think the reason is this. In the `Setup` func, it was originally written `close (c.ready)`, so need to renew the ready channel, but I changed it to `c.ready <-true` in the setup, and I will try to It changed back.

File: input/kafka/inputkafka.go (hunk 3 above)
Comment (McEchean) [reply input/kafka/inputkafka.go]: changed to main context

File: input/kafka/inputkafka.go (hunk 2 above)
Comment (tsaikd) [reply input/kafka/inputkafka.go]: I guess the example assumes you would reuse the channel. But in this case, the `Start()` executes only once. Anyway, renew the channel might not break anything.

**Comment by tsaikd** [comment]
Could you squash the commits before I merge it?

**Comment by McEchean** [comment]
> Could you squash the commits before I merge it?

Ok, it's done

**Pull request merged**